### PR TITLE
Add pagination support for frontend

### LINKED ISSUE #181: Add pagination support for the frontend
**Is your feature request related to a problem? Please describe.**
the backend already supports pagination, but the frontend currently does not utilize this feature, leading to performance issues when loading large datasets.

**Describe the solution you'd like**
Update the existing TanStack Query hooks to handle pagination by leveraging the backend’s pagination support.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Madhawa97**
## Purpose

The purpose of this PR is to fix #181

## Goals

The pagination support for the backend was recently added and this PR updates the relevant endpoints in the frontend to support the feature.

##  Checklist
- [ ] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [ ] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/pages/Mentors/index.tsx [hunk 1]
```diff
@@ -1,33 +1,67 @@
...
-  } = usePublicMentors(selectedCategory);
-  const {
-    data: categories,
+    data: categoriesData,
     isLoading: categoriesLoading,
     error: categoriesError,
-  } = useCategories();
+    fetchNextPage: fetchNextCategories,
+    hasNextPage: hasNextCategoriesPage,
+  } = useCategories(pageSize);
```
Comment (anjula-sack) [inline src/pages/Mentors/index.tsx]: Remove pagination for this @Madhawa97

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: src/pages/MenteeRegistration/index.tsx [hunk 2]
```diff
@@ -392,12 +386,12 @@ const MenteeRegistration: React.FC = () => {
...
           </>
         )}
-        {mentorsError !== null ? (
+        {status === 'error' ? (
           <div
             className="p-4 mb-4 text-sm text-red-800 rounded-lg bg-red-50 "
             role="alert"
           >
-            {mentorsError.message}
+            {'An error occurred. Please try again later.'}
```
Comment (anjula-sack) [inline src/pages/MenteeRegistration/index.tsx]:
```suggestion
            An error occurred. Please try again later.
```

**Review by anjula-sack — CHANGES_REQUESTED** [review]

**Comment by Madhawa97** [comment]
Updated to match with backend pagination removal for `categories`

We may need a new approach to get the `Total Mentee Slots: X` count in the admin mentor applications page.
Also the `totalItemCount` sends when filtered by category seems wrong and should be fixed in the backend.
@anjula-sack

**Comment by anjula-sack** [comment]
> Updated to match with backend pagination removal for `categories`
> 
> We may need a new approach to get the `Total Mentee Slots: X` count in the admin mentor applications page. Also the `totalItemCount` sends when filtered by category seems wrong and should be fixed in the backend. @anjula-sack

Can you create an issue for this?

**Review by anjula-sack — APPROVED** [review]

**Pull request merged**